### ISSUE TITLE
Low `flushInterval` causes console App object creation to error out

### ISSUE THREAD

**Issue by JoshCoady** [craftcms/cms#9818]
### Description

This is happening because the custom target in the config has not yet been converted into a Target object when something is tried to be logged during bootstrapping of the app.

This line calls `$target->enabled` but it assumes `$target` is a Target object where in the case it is an array.
https://github.com/yiisoft/yii2/blob/b5b1531935e43b731966928e2260530e7b5f0aa7/framework/log/Dispatcher.php#L190

This is because init has not yet been called on the dispatcher at this point
https://github.com/yiisoft/yii2/blob/b5b1531935e43b731966928e2260530e7b5f0aa7/framework/log/Dispatcher.php#L96

Which happens because the Craft dispatcher is calling `App::defaultLogTargets()` before it calls the yii dispatcher init
https://github.com/craftcms/cms/blob/ede3fd30c87ddc42057fa6a54cffaf12fa9dfc00/src/log/Dispatcher.php#L34

Which calls `Craft::$app->getIsInstalled()` if devMode is disabled
https://github.com/craftcms/cms/blob/374f6690ea7d56f9ccdad38c4b9a56dd780bff42/src/helpers/App.php#L639

Which checks for a valid DB connection
https://github.com/craftcms/cms/blob/374f6690ea7d56f9ccdad38c4b9a56dd780bff42/src/base/ApplicationTrait.php#L312

Which opens a DB connection
https://github.com/craftcms/cms/blob/374f6690ea7d56f9ccdad38c4b9a56dd780bff42/src/base/ApplicationTrait.php#L832

Which attempts to log a message
https://github.com/yiisoft/yii2/blob/7966fe47e6cbf0e87bb552ab60be173e9dd30d2a/framework/db/Connection.php#L631

But we are in the middle of initializing the logger, so it is not fully ready to log messages, so when it tries to dispatch the message because the flush interval is low, it produces the error.

### Steps to reproduce

1. In the config/app.php set the log component to have a flushInterval between 1 and 4 and also configure a custom target
```
        'log' => [
            'flushInterval' => 1,
            'targets' => [
                [
                    'class' => StreamLogTarget::class,
                    'url' => 'php://stderr',
                    'levels' => ['error', 'warning'],
                ],
            ],
        ],
```
2. Make sure you are running with devMode disabled
3. Try to run any command, for example `craft help`

This results in:
```
$ craft help
Error: Trying to get property 'enabled' of non-object
```

This is a screenshot of a debugging session showing the full stack trace and that $target is an array and not a Target object. If I take one more step at this point the error is triggered.
![image](https://user-images.githubusercontent.com/168141/132912634-c4fb4191-50fa-4961-baa9-0eef1ece8bf1.png)

### Additional info

- Craft version: 3.7.12
- PHP version: 
- Database driver & version:
- Plugins & versions:

**Comment by JoshCoady** [comment]
Also, it seems like there is no way to get rid of the default craft log targets since they are merged onto any additional target config?

**Comment by brandonkelly** [comment]
Thanks for reporting that! Just released Craft 3.7.13 with a fix.



> Also, it seems like there is no way to get rid of the default craft log targets since they are merged onto any additional target config?

Correct.
